risc0/zkvm/src/host/server/exec/tests.rs

```diff
@@ -738,6 +766,23 @@ fn args() {
     }
 }
 
+#[test]
+fn buf_read() {
+    let input = b"1234567";
+    let env = ExecutorEnv::builder()
+        .env_var("TEST_MODE", "BUF_READ")
+        // Previously failed on anything > buf and not % 4 == 0
```

Comment:
It's worth referencing this PR for more context.
```suggestion
        // Previously failed on anything > buf and not % 4 == 0
        // https://github.com/risc0/risc0/pull/1557
```